pkg/machinepool/machinepool.go

```diff
@@ -2118,6 +2142,23 @@ func getNodePoolReplicas(cmd *cobra.Command,
 		IsHostedCp:     true,
 	}
 
+	// Get capacity reservation limits for interactive mode
+	var capacityReservationId string
+	var availableInstances int32
+	if interactive.Enabled() && nodePool != nil && nodePool.AWSNodePool() != nil &&
+		nodePool.AWSNodePool().CapacityReservation() != nil {
+		capacityReservationId, _ = nodePool.AWSNodePool().CapacityReservation().GetId()
+		if capacityReservationId != "" && r != nil && r.AWSClient != nil {
```

Comment:
In what situation would `r` be nil here?

Reply:
Looking at the code, I'm not entirely sure when `r` would be nil, I went ahead and changed this because function already uses `r` earlier, so if `r` was nil, the code would have crashed before, same goes for the AWS client, my bad on this one.

Reply:
@olucasfreitas if you look into the codebase, you will see `r`'s values are already checked to be nil in the `runner` for the CLI. Before a single command is ran, this is already validated